### ISSUE TITLE
Use of Unsafe in High Scale break on Android

### ISSUE THREAD

**Issue by donv** [jruby/jruby#3554]
Using jruby-jars 1.7.23 with Ruboto results in this exception:

https://travis-ci.org/ruboto/ruboto/jobs/97978906#L4896 :

```
W/System.err(  870): Caused by: java.lang.RuntimeException: Could not obtain access to sun.misc.Unsafe
W/System.err(  870): 	at org.cliffc.high_scale_lib.UtilUnsafe.getUnsafe(UtilUnsafe.java:25)
W/System.err(  870): 	at org.cliffc.high_scale_lib.NonBlockingHashMapLong.<clinit>(NonBlockingHashMapLong.java:88)
W/System.err(  870): 	... 91 more
W/System.err(  870): Caused by: java.lang.NoSuchFieldException: theUnsafe
W/System.err(  870): 	at java.lang.Class.getDeclaredField(Class.java:631)
W/System.err(  870): 	at org.cliffc.high_scale_lib.UtilUnsafe.getUnsafe(UtilUnsafe.java:21)
```

JRuby uses Unsafe already, so I am not sure what the problem is with High Scale.  Any help would be great.

**Comment by mkristian** [comment]
another high scale issue is #3555

**Comment by kares** [comment]
seems to me that this pretty much nails that we're better off extracting the `NonBlockingHashMapLong` class into our own and managing the missing Unsafe. // cc @headius

**Comment by kares** [comment]
one solution is to port over the needed class with adjustments: https://github.com/jruby/jruby/issues/3560 ... and avoid packaging the .jar (which also resolves packaging patched JDK classes - #3555)

**Comment by donv** [comment]
@kares Thanks!
